astropy/coordinates/sky_coordinate.py

```diff
@@ -1147,14 +1150,19 @@ def _get_units(args, kwargs):
     return units
 
 
-def _parse_coordinate_arg(coords, frame, units):
+def _parse_coordinate_arg(coords, frame, units, scinit_kwargs):
```

Comment:
I think ``kwargs`` isn't good because it makes it seem like it's kwargs to this function, rather than the `SkyCoord`.  But you're right `init_kwargs` makes more sense so I'll do that.